Hestia/app/src/main/java/hestia/backend/PluginInformationRetrieverTask.java

```diff
@@ -0,0 +1,79 @@
+package hestia.backend;
+
+import java.util.HashMap;
+
+import android.app.Activity;
+import android.os.AsyncTask;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+
+import hestia.UI.AddDeviceInfo;
+
+public class PluginInformationRetrieverTask extends AsyncTask<Void,Void,HashMap<String,String>> {
+    private static final String TAG = "PluginRetrieverTask";
+    private String path;
+    private Activity a;
+
+    public PluginInformationRetrieverTask(String path, Activity a) {
+        this.path = path;
+        this.a = a;
+    }
+
+    @Override
+    protected HashMap<String, String> doInBackground(Void... params) {
+        URL url = null;
+        HttpURLConnection urlConnection = null;
+        HashMap<String,String> plugins = null;
+        try {
+            url = new URL(path);
+            urlConnection = (HttpURLConnection) url.openConnection();
+            InputStream input = new BufferedInputStream(urlConnection.getInputStream());
+            Log.i(TAG, input.toString());
+            plugins = readStream(input);
+
+        } catch (IOException e) {
+            Log.e(TAG, e.toString());
+        } finally {
+            if(urlConnection != null) {
+                urlConnection.disconnect();
+            }
+        }
+        Log.i(TAG, "PLUGINS: " + (plugins != null ? plugins.toString() : "NULL"));
+        return plugins;
+    }
+
+    @Override
+    protected void onPostExecute(HashMap<String,String> plugins) {
```

Comment:
It is rather difficult to make it inside the class. but we can try and automate the refresh system so that you don't always have to manually refresh it.